Feature/flexible terminal colour

beinvisible opened the pull request:
This is a rework/cleanup of (the first part of) efun::terminal_colour().

It allows to configure the matching strategy for colour keys by changing the delimiter string and using variable-length (enclosed by two delimiters) or fixed-length (following a single delimiter) colour codes. The configuration can either be done at compile time using --with-tc-delimiter and --with-tc-code-len, on startup with corresponding CLI parameters, or at runtime with configure_driver(DC_TERMINAL_COLOUR, ({ <"delimiter">, <code_len> })).

It removes support for the "%%^^" variant of escaping the colour key though, as this was a major clutter in the code. Support can easily be restored by using the provided sefun in mudlib/deprecated/ if someone really needs this variant.

beinvisible committed:
update beutelland settings

beinvisible committed:
rework of efun::terminal_colour()
    make colour key matching configurable

beinvisible commented:
Update: Change is running in Beutelland since 03 Apr 2022 18:17:55 - so far no issues (and much faster colorization because we don't have to convert to the old format anymore).

amotzkau commented:
As this is a breaking change, it wouldn't go into LDMud before 3.7 (which is not yet on the horizon).

Beside that, we're not sure, that it's a good idea to make major behavioral quirks of efuns configurable. There are only a few examples of that in the current driver (compat mode, regex engine selection) and I would rather like to get rid of them, than to expand on them. (Actually compat mode will be gone in 3.7.)

beinvisible commented:
I guess the configure_driver() option might be a bit overkill (after all that's something you'd rarely ever want to change at runtime anyways), but what is the problem with having at least the compile-time option?

beinvisible commented:
Alternatively I could implement the whole thing via an optional argument to terminal_colour(). Basically just directly pass the array used in configuire_driver(). That would completely get rid of the configuration options.

beinvisible commented:
Ok, I removed the whole driver-config stuff and replaced it with additional parameters to terminal_colour().

Testing in Beutelland, but since the main code did not change I don't expect any problems.

Mehran33SP commented on src/settings/beutelland:
_Deleted my account_

beinvisible committed:
add configurable end-delimiter for key matching

beinvisible commented:
Since it was easily done I added the option to also configure the end-delimiter for color codes separately.
So terminal_colour() now also supports variable-length color codes like e.g. `"<red>red text<green>green text"`

beinvisible commented:
rebased to 3.6.8

beinvisible committed:
update parameter description
